### PR TITLE
varlink: make comments on enum entries work

### PR DESCRIPTION
**Pull request by poettering**
Seems I never implemented that part correctly. Fix that. And the include appropriate comments for the io.systemd.BootControl Varlink interface so that this is thoroughly tested.

### CODE REVIEW

**Comment by poettering** [comment]
Follow-up for: fbb69c0306d434153ca85b227c7d42b9c92872e4

File: src/shared/varlink-idl.c [hunk 1]
```diff
@@ -74,28 +74,93 @@ static int varlink_idl_format_comment(
...
                 if (first) {
                         first = false;
                         fputs("(\n", f);
                 } else
                         fputs(",\n", f);
 
-                fputs(strempty(indent), f);
-                fputs("\t", f);
+                /* We found an enum value we want to output. In this case, output all immediately preceding
+                 * comments first. First, find the first comment in the series before. */
```
Comment (yuwata) [inline src/shared/varlink-idl.c]: first first first :-)

File: src/boot/bootctl-status.c [hunk 2]
```diff
@@ -886,6 +889,8 @@ int vl_method_list_boot_entries(Varlink *link, sd_json_variant *parameters, Varl
                         return r;
         }
 
-        return varlink_replyb(link, SD_JSON_BUILD_OBJECT(
-                                              SD_JSON_BUILD_PAIR_CONDITION(!!previous, "entry", SD_JSON_BUILD_VARIANT(previous))));
+        if (!previous)
+                return varlink_error(link, "io.systemd.BootControl.NoSuchBootEntry", NULL);
+
+        return varlink_replyb(link, SD_JSON_BUILD_OBJECT(SD_JSON_BUILD_PAIR("entry", SD_JSON_BUILD_VARIANT(previous))));
```
Comment (yuwata) [inline src/boot/bootctl-status.c]: `SD_JSON_BUILD_PAIR_VARIANT("entry", previous)` is shorter.

**Review by yuwata — APPROVED** [review]

File: src/shared/varlink-idl.c
Comment (yuwata) [inline src/shared/varlink-idl.c:1]: Maybe, better to add a test case for format comments for enums?

File: src/shared/varlink-idl.c [hunk 3]
```diff
@@ -74,28 +74,94 @@ static int varlink_idl_format_comment(
...
                 if (first) {
                         first = false;
                         fputs("(\n", f);
                 } else
                         fputs(",\n", f);
 
-                fputs(strempty(indent), f);
-                fputs("\t", f);
+                /* We found an enum value we want to output. In this case, start by outputting all
+                 * immediately preceeding comments. For that find the first comment in the series before the
```
Comment (yuwata) [inline src/shared/varlink-idl.c]: typo: preceeding -> preceding

File: src/shared/varlink-idl.c [hunk 4]
```diff
@@ -245,24 +311,15 @@ static int varlink_idl_format_all_fields(
...
-                for (const VarlinkField *c1 = field; c1 > symbol->fields; c1--) {
-                        const VarlinkField *c0 = c1 - 1;
-
-                        if (c0->field_type != _VARLINK_FIELD_COMMENT)
-                                break;
-
-                        start_comment = c0;
-                }
+                /* We found a field we want to output. In this case, start by outputting all immediately
+                 * preceeding comments. For that find the first comment in the series before the field, so
```
Comment (yuwata) [inline src/shared/varlink-idl.c]: Also here.

**Review by yuwata — APPROVED** [review]

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**